providers/aws/resources/aws.lr

```diff
@@ -2257,6 +2263,128 @@ aws.ec2 {
   networkAcls() []aws.ec2.networkacl
   // List of keypairs for the account
   keypairs() []aws.ec2.keypair
+  // List of EIPs
+  eips() []aws.ec2.eip
+}
+
+// Amazon Elastic IP
+private aws.ec2.eip {
+  // PublicIP assigned to 
+  publicIp string
+  // false if allocationId not present
+  attached bool 
+  // Ec2 instance associated with the EIP
+  instance() aws.ec2.instance
+  // Id for the network interface
+  networkInterfaceId string
+  // Id for the network interface owner
+  networkInterfaceOwnerId string
+  // Private IP address for the EIP
+  privateIpAddress string
+  // Ipv4 pool for the EIP
+  publicIpv4Pool string
+  // Tags for the EIP
+  tags map[string]string
+  // Region where the EIP is located
+  region string
+}
+
+// Amazon VPC NAT Gateway
+private aws.vpc.natgateway {
+  // Time when the gateway was created
+  createdAt time 
+  // Id for the NAT gateway
+  natGatewayId string
+  // State of the NAT gateway (pending | failed | available | deleting | deleted)
+  state string
+  // Tags for the NAT gateway
+  tags map[string]string
+  // VPC associated with the NAT gateway
+  vpc() aws.vpc
+  // List of addresses associated with the NAT gateway
+  addresses []aws.vpc.natgateway.address
+}
+
+// Amazon VPC NAT gateway address
+private aws.vpc.natgateway.address {
+  // Allocation Id for the address
+  allocationId string
+  // Network interface Id for the address
+  networkInterfaceId string
+  // Private IP associated with the address
+  privateIp string
+  // EIP associated with the address
+  publicIp() aws.ec2.eip // AllocationId can get us back to the actual EIP
+  // True if address is the primary one for the NAT gateway
+  isPrimary bool
+}
+
+// Amazon VPC Service Endpoint
+private aws.vpc.serviceEndpoint {
+  // True if acceptance required
```

Comment:
```suggestion
  // Whether acceptance is required
```
I'd love to finish this thought -- required for for what to happen with this endpoint?

Reply:
from the api
```
	// Indicates whether VPC endpoint connection requests to the service must be
	// accepted by the service owner.
```